### ISSUE TITLE
Couldn't get https://my.freenom.com/dologin.php

### ISSUE THREAD

**Issue by majatocaha** [maxisoft/Freenom-dns-updater#11]
Noticed this today. When the script runs, I get the following error message:

```
Traceback (most recent call last):
  File "/usr/local/bin/fdu", line 11, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/freenom_dns_updater/scripts/fdu.py", line 227, in update
    return _update(config, ignore_errors)
  File "/usr/local/lib/python2.7/dist-packages/freenom_dns_updater/scripts/fdu.py", line 199, in _update
    ok_count, err_count = record_action(lambda freenom, rec: freenom.add_record(rec, True), config, ignore_errors)
  File "/usr/local/lib/python2.7/dist-packages/freenom_dns_updater/scripts/fdu.py", line 235, in record_action
    if not freenom.login(config.login, config.password):
  File "/usr/local/lib/python2.7/dist-packages/freenom_dns_updater/freenom.py", line 38, in login
    assert r, "couldn't get %s" % url
AssertionError: couldn't get https://my.freenom.com/dologin.php
```

Could you look into this?

**Comment by fr1sk** [comment]
Have exactly the same error
`Traceback (most recent call last):
  File "/usr/local/bin/fdu", line 11, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/freenom_dns_updater/scripts/fdu.py", line 156, in record_update
    ok_count, err_count = record_action(lambda freenom, rec: freenom.add_record(rec, True), config, False)
  File "/usr/local/lib/python2.7/dist-packages/freenom_dns_updater/scripts/fdu.py", line 235, in record_action
    if not freenom.login(config.login, config.password):
  File "/usr/local/lib/python2.7/dist-packages/freenom_dns_updater/freenom.py", line 38, in login
    assert r, "couldn't get %s" % url
AssertionError: couldn't get https://my.freenom.com/dologin.php`

**Comment by echotrue90** [comment]
Need updating

**Comment by wangjiyang** [comment]
I also have this issue.

**Comment by uldics** [comment]
Their site shows this address as current with logon fields: https://my.freenom.com/clientarea.php
When changing the only place in the errors last mentioned file to https://my.freenom.com/clientarea.php, I get error `Unable to login with the given credential`. That didn't help, changed it back.

**Comment by xand79** [comment]
I'm drilling my own .sh csript with curl (it doesn't work too) and something gives me cue that it's because of temporary .pam solution and we must dig around freenom's TLS. Or not?

**Comment by echotrue90** [comment]
..

**Comment by Shihira** [comment]
Found a workaround. I posted the dologin.php along with some additional headers, and it just worked.

Edit freenom.py:37 or find the line that posts dologin.php, change it to:
```
r = self.session.post(url, playload, headers={'Host': 'my.freenom.com', 'Referer': 'https://my.freenom.com/clientarea.php'})
```

**Comment by ToJIka4** [comment]
Well finally it works! Thanks to @Shihira for fix script.
But most hard part of my problem was in SSL. Just append _COMODORSADomainValidationSecureServerCA.crt_ content into `certifi/cacert.pem`
I think it's temporary solution

**Comment by J-Cake** [comment]
@Shihira Thanks so much!!! it worked

**Comment by maxisoft** [comment]
Should be fixed by #23
